packages/mask/src/plugins/RedPacket/SNSAdaptor/NftRedPacketHistoryItem.tsx

```diff
@@ -146,7 +146,7 @@ export const NftRedPacketHistoryItem: FC<NftRedPacketHistoryItemProps> = memo(
         )
 
         useEffect(() => {
-            if (entry?.isIntersecting && rpid) setVisible(true)
+            if (entry?.isIntersecting && rpid) setIsViewed(true)
```

Comment:
`useEffect` is redundant.

```ts
const isViewed = entry?.isIntersecting && rpid
```

Reply:
`entry?.isIntersecting` can change from `true` to `false`.